### ISSUE TITLE
Create IdealLap Value Object for Reference Lap Data

### ISSUE THREAD

**Issue by yannicktuerk** [yannicktuerk/F1-Lap-Bot#44]
## Summary
Define IdealLap VO containing ideal speed profile and reference points for comparison with actual laps. Used by Mathe-Coach to compute time deltas and identify improvement areas.

## Technical Details
Create `IdealLap` VO in `src/domain/entities/ideal_lap.py`:

```python
@dataclass(frozen=True)
class IdealLap:
    track_id: str
    ideal_speed: np.ndarray        # Speed at each distance point (m/s)
    ideal_throttle: np.ndarray     # Binary throttle input (0/1)
    ideal_brake: np.ndarray        # Binary brake input (0/1)
    distance: np.ndarray           # Distance array aligned with TrackProfile
    sector_times: List[float]      # Ideal sector times [s1, s2, s3]
    total_time: float              # Total ideal lap time (seconds)
    
    def get_speed_at(self, distance: float) -> float:
        """Interpolate ideal speed at given distance."""
        return float(np.interp(distance, self.distance, self.ideal_speed))
    
    def get_time_loss_at(self, distance: float, actual_speed: float) -> float:
        """
        Compute instantaneous time loss at distance.
        
        Time loss = (1/v_actual - 1/v_ideal) * ds
        """
        ideal_speed = self.get_speed_at(distance)
        if ideal_speed > 0 and actual_speed > 0:
            return (1.0 / actual_speed - 1.0 / ideal_speed)
        return 0.0
    
    def compute_total_time_loss(self, actual_lap: 'LapTrace') -> float:
        """
        Compute total time loss over lap.
        """
        total_loss = 0.0
        for sample in actual_lap.samples:
            dt_loss = self.get_time_loss_at(sample.lap_distance, sample.speed)
            # Approximate ds
            ds = 10.0  # meters (sample spacing)
            total_loss += dt_loss * ds
        return total_loss
```

## Checklist
- [ ] Create IdealLap dataclass with speed/input profiles
- [ ] Implement query methods for distance-based lookups
- [ ] Add time loss calculation method
- [ ] Compute sector times from speed profile
- [ ] Validate array consistency and distance alignment
- [ ] Write unit tests for IdealLap construction and queries
- [ ] Test interpolation accuracy

## Acceptance Criteria
- IdealLap correctly encapsulates ideal lap data
- Supports delta calculations for Mathe-Coach feedback
- Speed interpolation accurate (<1% error)

## References
- Spec: `warp_prompt_f1_lap_bot.md` Section 5.4
